### ISSUE TITLE
avoid_bool_literals_in_conditional_expressions false positives

### ISSUE THREAD

**Issue by davidmorgan** [dart-lang/linter#1853]
If either side of the conditional expression is nullable, the suggested replacement fails.

```
condition ? true : myNullableFunction()
```

cannot be replaced with

```
condition || myNullableFunction()
```

--we could tighten up the lint to only apply to expressions that we know are not nullable, or we could wait for NNBD which will allow the lint to still fire on more cases.

While I like this lint I don't think it's terribly urgent, so my suggestion is to wait for NNBD.

What do you think, please?

**Comment by bwilkerson** [comment]
I agree that we should wait for NNBD. The analyzer currently has no way of determining whether an expression is nullable, so fixing it now would be difficult at best.

**Comment by davidmorgan** [comment]
Well, there are expressions that must evaluate to either `true` or `false`, e.g.:

```
condition ? true : foo == bar
```

--and this seems to be more common than nullable expressions. So it would still be quite useful. 

But changing it now just to change it again seems less good overall than awaiting for NNBD :)

Thanks.

**Comment by srawlins** [comment]
I'll close this in favor of null safety. If I've misunderstood, please re-open.
